Adaptors/SQS/src/PushQueueStorage.cs

```diff
@@ -90,36 +91,67 @@ await messages.GroupBy(m => m.Options.Priority)
-                                       logger_.LogDebug("pushed {messages} onto {QueueUrl}, {statusCode}, {responseMetadata}",
-                                                        entriesList.Select(entry => entry.Id)
-                                                                   .ToList(),
-                                                        queueUrl,
-                                                        response.HttpStatusCode,
-                                                        response.ResponseMetadata);
-                                     }
-
-                                     if (response.Failed.Any())
-                                     {
-                                       logger_.LogWarning("failed messages : {failed}",
-                                                          response.Failed.Select(entry => new
+                                       retry++;
+                                       var response = await client_.SendMessageBatchAsync(new SendMessageBatchRequest
+                                                                                          {
+                                                                                            QueueUrl = queueUrl,
+                                                                                            Entries  = entriesList,
+                                                                                          },
+                                                                                          cancellationToken)
+                                                                   .ConfigureAwait(false);
+
+                                       if (logger_.IsEnabled(LogLevel.Debug))
+                                       {
+                                         logger_.LogDebug("pushed {Messages} for {TaskIds} onto {QueueUrl}, {statusCode}, {responseMetadata}",
+                                                          entriesList.Select(entry => entry.Id)
+                                                                     .ToList(),
+                                                          entriesList.Select(entry => entry.MessageBody)
+                                                                     .ToList(),
+                                                          queueUrl,
+                                                          response.HttpStatusCode,
+                                                          response.ResponseMetadata);
+                                       }
+
+                                       if (response.Failed.Any())
+                                       {
+                                         var failed      = response.Failed.ToDictionary(entry => entry.Id);
+                                         var entriesDict = entriesList.ToDictionary(entry => entry.Id);
+
+                                         // there is at most 10 elements in this list
```

Comment:
Not related to this PR, but why did we chose to do groups of 10? I mean where does the 10 come from

Reply:
SQS support batches up to 10 messages